config/default/test/form-inputs.js

```diff
@@ -216,14 +216,20 @@ module.exports = {
         additionalFactors: ['yes', 'underweight']
       };
       Object.assign(content, opts);
+
+      const pregnancyAnswers = [content.firstPregnancy];
+      if (content.firstPregnancy === 'no') {
```

Comment:
```suggestion
      if (content.miscarriages) {
```

Instead of baking in some business logic here, can we just check for a `firstPregnancy` value?  I don't think we want the test util code to auto-correct the given answers if someone tries to provide both `firstPregnancy === 'no'` AND a value for `miscarriages`.  Instead the test should fail and they will need to update the provided answers.

Reply:
Hi @jkuester your suggestion will not work. It will allow the miscarriage question to be included based on the presence of a miscarriage value, regardless of whether firstPregnancy is 'yes' or 'no'.

For context, this is what we want: 

- The miscarriage question should only be asked if this is not the first pregnancy (firstPregnancy === 'no').
- If it is the first pregnancy (firstPregnancy === 'yes'), the miscarriage question should not be included at all.

Reply:
Right, I think we are on the same page here except that I am suggesting we make the test code _less smart_!  It is invalid to try and set `firstPregnancy === 'yes'` and `miscarriages === 'yes'` and now the form will actually not even let you fill out the miscarriage question when `firstPregnancy === 'yes'`.  However, as this test util code is currently written, I could still write a test where I say  `firstPregnancy: 'yes'` AND `miscarriages: 'yes'`.  Then, when I call this util code, it will just ignore my value for `miscarriages` and my test will pass even though I set bad data (then this can get confusing to understand and debug in the future).  Instead, I would rather just have this util code pass through whatever the data is provided by the test and just let it fail later if something is wrong (could alternatively just actively throw an exception here if we see bad data, but that is just more code to maintain....)

Reply:
I get it now.

However, I don't think `if (content.miscarriages) {}` will work as it is because it will attempt to fill in miscarriages even when `firstPregnancy` is `yes` So I think we can go with `if (content.firstPregnancy === 'no' && content.miscarriages) {}`

I believe that will keep the test utility function `simple` by not including logic to correct or ignore data but still respects the form's logic.

Reply:
Hmm, no, if `content.firstPregnancy === 'no'` then they have to provide a value for `miscarriages` to be able to complete the form.  So, checking them both is redundant!  Ultimately, this is just a super minor nitpick and I am good with what you had originally (`if (content.firstPregnancy === 'no')`) if that makes the most sense to you. Let's not let this hold up these changes!

That being said, I will make one more attempt and explaining myself better here with a bit more context!  :smile: 

The automated tests for these forms (e.g. the `pregnancy` form) use the cht-conf-test-harness to fill out the form in a test environment.  The way the `harness.fillForm` function works by taking an array of "answers" to questions on the form.  There is no mapping for which answer goes to which question in the form, so everything is just sequential. The first answer in the array goes to the first _visible_ question.  As you fill out a form, sometimes questions become visible (or get hidden) depending on what inputs you are providing. The test harness will always check for the next _visible_ question to see what it should fill out. If there are answers left at the end of the form page and all the questions have been filled out, then the harness will throw an exception saying that the given answers were invalid.

The tests (e.g. in the `pregnancy.spec.js` file) use the data from this `form-inputs.js` file to fill out forms for various scenarios.  For more complex workflows it can be useful to have the test data from `form-inputs.js` be somewhat _dynamic_ to allow for reuse between similar scenarios.  So in this case we are looking at the `pregnancyRegistrationScenarios.riskDanger` function which will return a set of data for the pregnancy form that is useful for testing various "risks" workflows (and is getting reused in a number of tests). The test data returned from this function is passed directly to the test-harness which will blindly try to fill out the fields on the form one-after-another.

Since this `pregnancyRegistrationScenarios.riskDanger` function is reusable and the consuming tests can pass in various combinations of `content` values, I am trying to avoid a future scenario where a test improperly calls this function with bad content data like `{ firstPregnancy: yes, miscarriages: 'yes' }` (since you cannot answer "yes" to both of these questions.  If a test calls `pregnancyRegistrationScenarios.riskDanger` with content like that, I want the test to FAIL.  The trouble is that with your logic, I think the test will actually PASS!  (You can confirm this by updating the `shows summary when this is the first pregnancy for the woman and there are all the risk factors` test in `pregnancy.spec.js` to set `miscarriages: 'yes'`.  That is an invalid test case that should fail, but I think your logic here will allow it to pass...)

If, instead, we just checked `if (content.miscarriages) {` here, then I think the test would fail since it would include the `miscarriages` answer in the array of answers that gets returned from `pregnancyRegistrationScenarios.riskDanger`. Then that array of answers would be passed to the test harness which would try to fill out the questions on the form.  With `firstPregnancy: yes`, then the miscarriages question should not become visible in the form and the harness will throw an error saying that it cannot fill out the form because it has too many answers.

Reply:
Hi @jkuester thanks for the detailed response. 

To avoid holding this PR up. I will revert to what I had `(if (content.firstPregnancy === 'no'))`

However, I'd like to continue this conversation elsewhere because as much as I understand your concern, I don't see how to write a "passing" test with `if (content.miscarriages) {}`

If that sounds good, let me know and I will merge this PR. Thanks.

Reply:
:+1: Yes, that sounds good!